src/com/powem/inv/algos/DrugInteractionAlert.java

```diff
@@ -0,0 +1,272 @@
+//      //TEST END
+//
+//      // TEST
+//      assert alerts.size() == 1;
+//      //TEST END
+//    }
+//
+//    {
+//      Map<String, Map<String, String>> interactions = new HashMap<>();
+//      interactions.put("DrugA", Map.of("DrugB", "high", "DrugC", "medium"));
+//      interactions.put("DrugB", Map.of("DrugA", "high"));
+//      interactions.put("DrugC", Map.of("DrugA", "medium"));
+//
+//      DrugInteractionAlert alertSystem = new DrugInteractionAlert(interactions);
+//      List<String> administeredDrugs = Arrays.asList("DrugA", "DrugB", "DrugC");
+//      List<String> alerts = alertSystem.generateAlerts(administeredDrugs);
+//
+//      // TEST
+//      assert alerts.contains("DrugA interacts with DrugB (high)");
+//      //TEST END
+//
+//      // TEST
+//      assert alerts.contains("DrugA interacts with DrugC (medium)");
+//      //TEST END
+//
+//      // TEST
+//      assert alerts.size() == 2;
+//      //TEST END
+//    }
+//
+//    {
+//      Map<String, Map<String, String>> interactions = new HashMap<>();
+//      interactions.put("DrugA", Map.of("DrugD", "high"));
+//      interactions.put("DrugB", Map.of());
+//      interactions.put("DrugC", Map.of());
+//
+//      DrugInteractionAlert alertSystem = new DrugInteractionAlert(interactions);
+//      List<String> administeredDrugs = Arrays.asList("DrugA", "DrugB", "DrugC");
+//      List<String> alerts = alertSystem.generateAlerts(administeredDrugs);
+//
+//      // TEST
+//      assert alerts.isEmpty();
+//      //TEST END
+//
+//    }
+//
+//    {
+//      Map<String, Map<String, String>> interactions = new HashMap<>();
+//      interactions.put("DrugA", Map.of("DrugB", "high"));
+//
+//      DrugInteractionAlert alertSystem = new DrugInteractionAlert(interactions);
+//      List<String> administeredDrugs = Arrays.asList("DrugA", "DrugB", "DrugD");
+//      List<String> alerts = alertSystem.generateAlerts(administeredDrugs);
+//
+//      // TEST
+//      assert alerts.contains("DrugA interacts with DrugB (high)");
+//      //TEST END
+//
+//      // TEST
+//      assert alerts.size() == 1;
+//      //TEST END
+//    }
+//
+//    {
+//      Map<String, Map<String, String>> interactions = new HashMap<>();
+//      interactions.put("DrugA", Map.of());
+//
+//      DrugInteractionAlert alertSystem = new DrugInteractionAlert(interactions);
+//      List<String> administeredDrugs = Arrays.asList("DrugA");
+//      List<String> alerts = alertSystem.generateAlerts(administeredDrugs);
+//
+//      // TEST
+//      assert alerts.isEmpty();
+//      //TEST END
+//    }
+//
+//    {
+//      Map<String, Map<String, String>> interactions = new HashMap<>();
+//      interactions.put("DrugA", Map.of("DrugB", "high"));
+//
+//      DrugInteractionAlert alertSystem = new DrugInteractionAlert(interactions);
+//
+//      try {
+//        alertSystem.generateAlerts(null);
+//
+//        //TEST
+//        assert false : "Null administered drugs should throw IllegalArgumentException";
```

Comment:
just assert false, remove "Null administered drugs should throw IllegalArgumentException"

Reply:
example: ```//        try {
//            detector.addTemperatureRecord(null, 20.0);
//            assert false;
//        } catch (IllegalArgumentException e) {
//            assert true;
//        }```